collection/rancher/v2.x/rancher-pod-collector/rancher-pod-collector.sh

```diff
@@ -0,0 +1,217 @@
+  ${KUBECTL_CMD} get nodes -o wide > $TMPDIR/clusterinfo/get-node-wide 2>&1
+  ## Grabbing cattle-system items
+  mkdir -p $TMPDIR/cattle-system/
+  ${KUBECTL_CMD} get pods -n cattle-system -o wide > $TMPDIR/cattle-system/get-pods 2>&1
+  ${KUBECTL_CMD} get svc -n cattle-system -o wide > $TMPDIR/cattle-system/get-svc 2>&1
+  ${KUBECTL_CMD} get endpoints -n cattle-system -o wide > $TMPDIR/cattle-system/get-endpoints 2>&1
+  ## Grabbing kube-system items
+  mkdir -p $TMPDIR/kube-system/
+  ${KUBECTL_CMD} get configmap -n cattle-system cattle-controllers -o yaml > $TMPDIR/kube-system/get-configmap-cattle-controllers.yaml 2>&1
+
+}
+
+enable-debug() {
+
+  techo "Enabling debug for Rancher pods"
+  for POD in $(${KUBECTL_CMD} get pods -n cattle-system -l app=rancher --no-headers | awk '{print $1}');
+  do
+    if [ ! -z "${TRACE}" ]
+    then
+      techo "Pod: $POD `${KUBECTL_CMD} exec -n cattle-system -c rancher $POD -- loglevel --set trace`"
+    else
+      techo "Pod: $POD `${KUBECTL_CMD} exec -n cattle-system -c rancher $POD -- loglevel --set debug`"
+    fi
+  done
+
+}
+
+disable-debug() {
+
+  techo "Disabling debug for Rancher pods"
+  for POD in $(${KUBECTL_CMD} get pods -n cattle-system -l app=rancher --no-headers | awk '{print $1}');
+  do
+    techo "Pod: $POD `${KUBECTL_CMD} exec -n cattle-system -c rancher $POD -- loglevel --set debug`"
+  done
+
+}
+
+capture-logs() {
+
+  techo "Capturing debug logs from Rancher pods"
+  mkdir -p $TMPDIR/rancher-logs/
+  for POD in $(${KUBECTL_CMD} get pods -n cattle-system -l app=rancher --no-headers | awk '{print $1}');
+  do
+    techo "Pod: $POD"
+    ${KUBECTL_CMD} -n cattle-system logs -c rancher $POD > $TMPDIR/rancher-logs/$POD
+  done
+
+}
+
+pause(){
+ read -n1 -rsp $'Press any key to continue or Ctrl+C to exit...\n'
+}
+
+archive() {
+
+  FILEDIR=$(dirname $TMPDIR)
+  FILENAME="$(hostname)-$(date +'%Y-%m-%d_%H_%M_%S').tar"
+  tar --create --file ${FILEDIR}/${FILENAME} --directory ${TMPDIR}/ .
+  ## gzip separately for Rancher OS
+  gzip ${FILEDIR}/${FILENAME}
+
+  techo "Created ${FILEDIR}/${FILENAME}.gz"
+
+}
+
+cleanup() {
+
+  techo "Removing ${TMPDIR}"
+  rm -r -f "${TMPDIR}" >/dev/null 2>&1
+
+}
+
+help() {
+
+  echo "Rancher Pod Collector
+  Usage: rancher-pod-collector.sh [ -d <directory> -r <container runtime> -k KUBECONFIG -t -f ]
+
+  All flags are optional
+
+  -d    Output directory for temporary storage and .tar.gz archive (ex: -d /var/tmp)
+  -r    Override container runtime if not automatically detected (docker|k3s)
+  -k    Override the kubeconfig (ex: ~/.kube/custom)
+  -t    Enable tracve logs
+  -f    Force log collection if the minimum space isn't available"
+
+}
+
+timestamp() {
+
+  date "+%Y-%m-%d %H:%M:%S"
+
+}
+
+techo() {
+
+  echo "$(timestamp): $*"
+
+}
+
+while getopts ":d:r:k:fth" opt; do
+  case $opt in
+    d)
+      MKTEMP_BASEDIR="-p ${OPTARG}"
+      ;;
+    r)
+      RUNTIME_FLAG="${OPTARG}"
```

Comment:
What is this for?

Reply:
Removing, this was leftover from the log script.